app/pod/components/weather-card/component.ts

```diff
@@ -2,17 +2,19 @@ import Component from '@glimmer/component';
 import weatherIcons from 'weather-app/utils/weather-icon';
 
 export default class WeatherCardComponent extends Component {
-  get icon() {
-    var prefix = 'wi wi-';
-    var code = this.args.weatherData.weather[0].id;
-    var icon = weatherIcons[code].icon;
+  get icon(): string {
+    const prefix: string = 'wi wi-';
+    const code: number = this.args.weatherData.weather[0].id;
```

Comment:
one thing we do with args, which I find really cool is we make an interface in the same file for the args, then make the component conform to the interface like this

```
export interface SomeComponentArgs {
  thing?: string;
  anotherThing: number;
}

export default class SomeComponent extends Component<SomeComponentArgs> {
  ...
}
```

you will find when doing this you not only get auto complete for your args but the type casting to boot 🎉